ql/termstructures/yield/oisratehelper.cpp

```diff
@@ -99,6 +100,10 @@ namespace QuantLib {
         QL_REQUIRE(termStructure_ != 0, "term structure not set");
         // we didn't register as observers - force calculation
         swap_->recalculate();
+        for (Leg::const_iterator c = swap_->overnightLeg().begin();
```

Comment:
Shouldn't maybe this go before the `swap->recalculate()`? The swap might need the updated values of the coupons...

Reply:
You are right. This also worked, since the floating rate coupons notified the swap and it would recalculate next time it is queried. However it meant that the swap was calculated two times. I just removed the recalculate calls on the swap, then I get the old timings again. Thanks a lot for pointing me at this.